app/helpers/reactjs_helper.rb

```diff
@@ -65,6 +65,15 @@ def css_tags_for(requested_plugins)
     end
   end
 
+  def locale_js_tags
+    locale = FastGettext.locale
+    domains = ::Foreman::Plugin.all.map(&:gettext_domain).compact
```

Comment:
Not a blocker, but Ruby 2.7 has introduced `filter_map`. https://blog.saeloun.com/2019/05/25/ruby-2-7-enumerable-filter-map/ shows some nice examples. Perhaps that can replace the `.map.compact.select` with a single `filter_map`.

Reply:
I must say I'm not a huge fan of mixing filtering and transformation into a single step, but sure, done